torchaudio/datasets/vctk_092.py

```diff
@@ -0,0 +1,150 @@
+import os
+import warnings
+from typing import Any, Tuple
+
+import torchaudio
+from torch import Tensor
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import (
+    download_url,
+    extract_archive,
+    walk_files
+)
+
+URL = "https://datashare.is.ed.ac.uk/bitstream/handle/10283/3443/VCTK-Corpus-0.92.zip"
+FOLDER_IN_ARCHIVE = "VCTK-Corpus"
+_CHECKSUMS = {
+    "https://datashare.is.ed.ac.uk/bitstream/handle/10283/3443/VCTK-Corpus-0.92.zip":
+    "8a6ba2946b36fcbef0212cad601f4bfa"
+}
+
+
+def load_vctk_item(fileid: str,
+                   path: str,
+                   ext_audio: str,
+                   ext_txt: str,
+                   folder_audio: str,
+                   folder_txt: str,
+                   downsample: bool = False) -> Tuple[Tensor, int, str, str, str]:
+    speaker_id, utterance_id, _ = fileid.split("_")
+
+    # Read text
+    #
+    # After 30th Nov 2018 update, a new mic was used to record some audio files.
+    # In the process all audio file names were changed to include the mic info.
+    #
+    # Old audio filename - p225_001
+    # New audio filename - p225_001_mic1
+    #
+    # So, using string slicing to use the appropriate fileid for text files.
+    file_txt = os.path.join(path, folder_txt, speaker_id, fileid[:-5] + ext_txt)
+
+    with open(file_txt) as file_text:
+        utterance = file_text.readlines()[0]
+
+    # Read flac
+    file_audio = os.path.join(path, folder_audio, speaker_id, fileid + ext_audio)
+    waveform, sample_rate = torchaudio.load(file_audio)
+    if downsample:
+        # TODO Remove this parameter after deprecation
+        F = torchaudio.functional
+        T = torchaudio.transforms
+        # rate
+        sample = T.Resample(sample_rate, 16000, resampling_method='sinc_interpolation')
+        waveform = sample(waveform)
+        # dither
+        waveform = F.dither(waveform, noise_shaping=True)
+
+    return waveform, sample_rate, utterance, speaker_id, utterance_id
+
+
+class VCTK_092(Dataset):
+    """
+    Create a Dataset for VCTK 0.92. Each item is a tuple of the form:
+    (waveform, sample_rate, utterance, speaker_id, utterance_id)
+    Folder `p315` will be ignored due to the non-existent corresponding text files.
+    For more information about the dataset visit: https://datashare.is.ed.ac.uk/handle/10283/3443
+    """
+
+    _folder_txt = "txt"
+    _folder_audio = "wav48_silence_trimmed"
+    _ext_txt = ".txt"
+    _ext_audio = ".flac"
+    _except_folder = "p315"
+
+    def __init__(self,
+                 root: str,
+                 url: str = URL,
+                 folder_in_archive: str = FOLDER_IN_ARCHIVE,
+                 download: bool = False,
+                 downsample: bool = False,
+                 transform: Any = None,
+                 target_transform: Any = None) -> None:
+
```

Comment:
Could you remove `downsample`, `transform` and `target_transform` parameters? Since we are adding a new class, we should take advantage of the opportunity not to carry out some legacy customs.

Reply:
can I add a new param - `mic_id` as there are recordings using a new mic from `0.92` version?

Reply:
I think you are referring this one

> 1: Audio files recorded via DPA 4035 (mic1) have very low frequency noises and therefore we added audio files recorded via MKH 800 (mic2)

I think that's reasonable to add the parameter. What is your thought on the behavior of `mic_id` ?
Assuming `mic_id` is integer type, I imagine giving either `1` or `2` would only retrieve the corresponding audio. Then I wonder if there is a use-case where one wants to randomize or get both of them. What do you think?

Reply:
It depends on several research use cases, if someone wants to do some research on only old audio with the low-frequency noises, it can be helpful. Besides users can conveniently ignore `mic_id`, currently, it's being ignored by default when slicing the filename in load item method.

Reply:
and one more thing, should I remove the deprecated params from the old `VCTK` class and `load_vctk_item` method too?

Reply:
> Besides users can conveniently ignore mic_id, currently, it's being ignored by default when slicing the filename in load item method.

My understanding is that microphone ID was ignored in the original `VCTK` implementation because there was only one ID there. Now that with 0.92 update, there are two microphones, and we would like to give users easy control and intuitive behavior. Randomizing microphone ID would require another parameter for random seed for the sake of reproducibility, and I think we can revisit that later.

Therefore, I think, for the sake of simplicity here, we can just allow `mic_id:int=1 or 2`, which fixes the micro phone ID. What do you think?

Reply:
> and one more thing, should I remove the deprecated params from the old `VCTK` class and `load_vctk_item` method too?

Let's keep the changes to the original `VCTK` minimum. So I suggest the only change we are going to make to the original `VCTK` class in the context of this PR will be raising error when `download=True`.

Reply:
> and one more thing, should I remove the deprecated params from the old `VCTK` class and `load_vctk_item` method too?

@mthrok

Reply:
okay, will push the changes after testing